src/MoBi.Presentation/Presenter/SelectFolderAndIndividualAndExpressionFromProjectPresenter.cs

```diff
@@ -0,0 +1,77 @@
+using System.Collections.Generic;
+using MoBi.Core.Domain.Repository;
+using MoBi.Presentation.DTO;
+using MoBi.Presentation.Mappers;
+using MoBi.Presentation.Presenter.BasePresenter;
+using MoBi.Presentation.Tasks.Edit;
+using MoBi.Presentation.Views;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Presentation.Presenters;
+
+namespace MoBi.Presentation.Presenter
+{
+   public interface ISelectFolderAndIndividualAndExpressionFromProjectPresenter : IDisposablePresenter
+   {
+      /// <summary>
+      ///    Opens a dialog where the user will select an IndividualBuildingBlock from the project and a file path to export
+      /// </summary>
+      (string, IndividualBuildingBlock, IReadOnlyList<ExpressionProfileBuildingBlock>) GetPathAndIndividualForExport(IContainer container);
+
+      /// <summary>
+      ///    Returns a list of all IndividualBuildingBlocks in the project
+      /// </summary>
+      IReadOnlyList<IndividualBuildingBlock> AllIndividuals { get; }
+
+      /// <summary>
+      ///    Opens a dialog for the user to select file path
+      /// </summary>
+      /// <returns>The path if dialog is dismissed with ok, empty string if canceled</returns>
+      string BrowseFilePath();
+
+      void SelectionChanged();
+   }
+
+   public class SelectFolderAndIndividualAndExpressionFromProjectPresenter : MoBiDisposablePresenter<ISelectFolderAndIndividualAndExpressionFromProjectView, ISelectFolderAndIndividualAndExpressionFromProjectPresenter>, ISelectFolderAndIndividualAndExpressionFromProjectPresenter
+   {
+      private readonly IBuildingBlockRepository _buildingBlockRepository;
+      private readonly IEditTaskForContainer _editTaskForContainer;
+      private readonly IObjectPathFactory _objectPathFactory;
+      private readonly IIndividualExpressionAndFilePathDTOMapper _mapper;
+      private IndividualExpressionAndFilePathDTO _dto;
+
+      public SelectFolderAndIndividualAndExpressionFromProjectPresenter(
+         ISelectFolderAndIndividualAndExpressionFromProjectView view, 
+         IBuildingBlockRepository buildingBlockRepository, 
+         IEditTaskForContainer editTaskForContainer, 
+         IObjectPathFactory objectPathFactory,
+         IIndividualExpressionAndFilePathDTOMapper mapper) : base(view)
+      {
+         _buildingBlockRepository = buildingBlockRepository;
+         _editTaskForContainer = editTaskForContainer;
+         _objectPathFactory = objectPathFactory;
+         _mapper = mapper;
+      }
+
+      public (string, IndividualBuildingBlock, IReadOnlyList<ExpressionProfileBuildingBlock>) GetPathAndIndividualForExport(IContainer container)
```

Comment:
Not found of this for public interfaces
Is it possible to name parameters? I think so
string path, IndividualBuildingblock individual etc..

Reply:
I'm going to name the tuple members for the PR. Is it enough, or rather have a class for this?